src/msg/Messenger.cc

```diff
@@ -22,10 +22,12 @@ Messenger *Messenger::create(CephContext *cct, const string &type,
 			     entity_name_t name, string lname,
 			     uint64_t nonce, uint64_t features)
 {
-  int r = -1;
-  srand(time(NULL));
-  if (type == "random")
+  static int r = -1;
```

Comment:
Please use a more descriptive variable name. We use "r" a lot as a return call and it would be sad if this code got refactored like that. :)